bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java

```diff
@@ -455,7 +456,9 @@ public void unblockStrand(Strand strand) {
     }
 
     private void cleanUp(Strand justCompleted) {
-        justCompleted.scheduler = null;
```

Comment:
Shall we just remove this line and do not make any other changes? Since we only create one Scheduler object to run the Ballerina program (and another one for module stop), the Scheduler object will not get garbage collected anyway. Hence, we do not have a reason to make this `null`. @warunalakshitha WDYT?

Reply:
If we are not making the reference of `justCompleted.scheduler` null, will that strand object be eligible for garbage collection even though it doesn't have no live references to other objects?

Reply:
AFAIK it will eligible to grabage collection since it is not reachable.